library/std/src/sys/windows/path.rs

```diff
@@ -141,3 +162,55 @@ fn parse_next_component(path: &OsStr, verbatim: bool) -> (&OsStr, &OsStr) {
         None => (path, OsStr::new("")),
     }
 }
+
+/// Gets an absolute path with a verbatim prefix.
+///
+/// This allows bypassing the legacy `MAX_PATH` limits.
+pub(crate) fn to_verbatim(path: &Path) -> io::Result<Vec<u16>> {
+    let mut path = to_u16s(path)?;
+
+    // Early return for paths that are already verbatim.
+    if path.starts_with(VERBATIM_PREFIX) || path.starts_with(NT_PREFIX) {
+        return Ok(path);
+    }
+
+    // Firstly, get the absolute path using `GetFullPathNameW`.
+    // https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getfullpathnamew
+    let lpfilename = path.as_ptr();
+    fill_utf16_buf(
+        // SAFETY: `fill_utf16_buf` ensures the `buffer` and `size` are valid.
+        // `lpfilename` is a pointer to a null terminated string that is not
+        // invalidated until after `GetFullPathNameW` returns successfully.
+        |buffer, size| unsafe { c::GetFullPathNameW(lpfilename, size, buffer, ptr::null_mut()) },
```

Comment:
Hmm, according to [this stackoverflow answer](https://stackoverflow.com/questions/38036943/getfullpathnamew-and-long-windows-file-paths) and the [this Google Blog](https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html) it should work though.

Reply:
I think that's automatic boiler plate. I've tested on Windows 7 to Windows 11 and none enforce `MAX_PATH`. I'm pretty sure it would be breaking code to change it now.

If `GetFullPathNameW()` cannot be used without `\\?\` then it would defeat the purpose of using it here. So it will be necessary to reimplement it in Rust.

Reply:
> Maybe add a comment that this is not according to doc but should work (TM).

Oh sorry, missed this as I was typing my last comment. Yes, this seems like a good idea. And I do have a test case that explicitly checks for long paths.